### DISCUSSION TITLE
Error: Invalid virtual machine configuration. The storage device attachment is invalid.

### DISCUSSION THREAD

**Discussion by LandeeAu** [utmapp/UTM#6622] [Linux]
Hi guys, 
 I am trying to configure Linux (jammy-desktop-arm64), I got Error: **Invalid virtual machine configuration. The storage device attachment is invalid**. as I click the button to start it.  My laptop is a MacBook Pro M1max. any advice on it?

**Comment by mjwSolver** [comment]
Hey, I know this response is very late, but I recently had a similar issue but it was with Linux Ubuntu 25.10 on UTM. Hope it helps anyone (including forgetful me) in the coming future who may have faced similar issues. 

_TLDR_; **We redownloaded the ISO file** and it worked on our end. Probably a corrupted download at the time. Sorry if it's not exactly a one-to-one match with your case and hope this helps anyone else in the future. 

---

Long story: We had two macbooks, an M2 and an M3. The M2 had UTM version 4.5.4 and after downloading Ubuntu 25.10 it worked perfectly fine on the M2 MacBook. For the M3 though, we tried downgrading from 4.7.4 to the same version the M2 had installed which was 4.5.4, even though we did, it still returned that same error; _confirming_ that isn't a "newer UTM version issue".

We narrowed down that the M2 works, so the default approach (add ISO, set RAM and Storage, Virtualize for Apple Silicon and so forth) is fine, so no extra configuration or setting change was actually needed. We thought some setting might have persisted where the "storage" setting was the issue (maybe M3 changed something and despite reinstalling a downgraded version it somehow persisted), but after just trying to redownload the ISO file, it fixed it and M3 was working. 

Note: **Sometimes corrupted files will look different on Mac (A good example: unfinished downloaded file), the icon would look like a blank txt document, but this wasn't the case for the corrupted copy of the ISO file on the M3 at the time**. 

Additional resources:
- Finding previous release versions of UTM: https://github.com/utmapp/UTM/releases

**Comment by biryusal** [comment]
I managed to resolve the error Invalid virtual machine configuration. The storage device attachment is invalid when using the Apple Virtualization Framework.

It seems to be a mounting/configuration glitch in how the framework handles the ISO attachment. The fix is simple:
1. Shut down the VM.
2. Delete the existing ISO drive from the Drives section.
3. Add a new drive, select Import, and pick the ISO again.
4. Ensure the "Read Only" checkbox is checked.

Even with the exact same settings as before, re-adding the drive manually seems to "refresh" the configuration and allows the VM to boot correctly.
